Add Contributors guidance for Dioptra

jkglasbrenner opened the pull request:
Closes #80 
Closes #81

hbooth commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,81 @@
+
+If you find a bug or unexpected behavior please use the provided [issue template](https://github.com/usnistgov/dioptra/issues/new/choose) and describe the issue along with step-by-step instructions on how to reproduce the issue.
+
+# Feature Requests
```
Perhaps be a sub-header (in other words needs another '#' character)?

hbooth commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,81 @@
+
+If you’re working locally, it often can be useful to `--amend` a commit, or utilize `rebase -i` to reorder, squash, and reword your commits.
+
+# Commit Style Guide
```
I think may also be a sub-heading too. Either that, or the next one should be a top-level header, although I think sub-heading may make more sense.

hbooth reviewed (changes requested):
Just the two comments and it should be ready to go.

jkglasbrenner commented:
@hbooth I looked at the Markdown headers you pointed out and agree they were at the wrong level. After looking at CONTRIBUTING.md again in light of your feedback, here's the new subject heading hierarchy:

- Criteria for Contributions and Feedback
- Contributor Responsibilities
- Question or Problem?
- Submitting Issues
    - Bugs
    - Feature Requests
- Code Submission Guidelines
    - Development Workflow
    - Commit Style Guide
        - Squashing
    - Code Reviewing Guide

I moved the squashing section under Commit Style Guide since that seemed to make more sense, and kept Code Reviewing Guide as a sub-heading up Code Submission Guidelines.

Also, I didn't notice that we had more than one `#` header in the file, the convention is to reserve `#` for the title and only use it once. I also fixed that up.

I think it looks good now, let me know if you agree and we can get this merged.

hbooth reviewed (approved):
Agreed and everything looks good to me.
